src/test_core.py

```diff
@@ -51,6 +51,17 @@ def test_override_default_config_with_user_config(self):
         self.assertFalse(final_config["mdx_configs"]["toc"]["permalink"])
         self.assertTrue("mdx_truly_sane_lists" in final_config["markdown_extensions"])
 
+    def test_remove_theme_config_with_user_options(self):
+        self.mkdocs_yaml_config["theme"]["features"] = ["navigation.sections"]
+        final_config = self.techdocscore.on_config(self.mkdocs_yaml_config)
+        self.assertFalse("navigation.sections" in final_config["theme"]["features"])
+
+    def test_override_theme_config_with_user_config(self):
+        self.mkdocs_yaml_config["theme"] = Theme(name="material")
+        self.mkdocs_yaml_config["theme"]["features"] = ["navigation.sections"]
+        final_config = self.techdocscore.on_config(self.mkdocs_yaml_config)
+        self.assertTrue("navigation.sections" in final_config["theme"]["features"])
+
     def test_template_renders__multiline_value_as_valid_json(self):
```

Comment:
I'd also love to see two more test cases:

- One related to how `config["theme"].static_templates` is handled (e.g. check that both the override value provided in config and the default value `techdocs_metadata.json` are present in the final config)
- Another related to how `config["theme"].dirs` is handled (e.g. check that both the override value provided in config and the default `self.tmp_dir_techdocs_theme.name` are present in the final config)